### PR TITLE
Disallow spectators to use most !commands

### PR DESCRIPTION
**Pull request by sprunk**
Technical and info commands are kept:

!spawn
!adduser
!transmit
!voteresign
!notify
!predict
!springie

### CODE REVIEW

**Comment by Licho1** [comment]
I think that votemove should also be allowed because it moves spec to another host.
Or !help..
And it was also purpose to have voterehost and votekick available due to extreme circumstances - like all people resigned except one lagger or host broken.

File: Springie/Springie/autohost/AutoHost.cs [hunk 1]
```diff
@@ -233,7 +238,7 @@ public void Start()
                     for (int i = 0; i < c.ListenTo.Length; i++) {
                         if (c.ListenTo[i] == e.Place) {
                             // command is only for nonspecs
-                            if (!c.AllowSpecs) if (tas.MyBattle == null || !tas.MyBattle.Users.Values.Any(x => x.LobbyUser.Name == e.UserName && !x.IsSpectator)) return false;
+                            if (!c.AllowSpecs && !GetUserAdminStatus(e)) if (tas.MyBattle == null || !tas.MyBattle.Users.Values.Any(x => x.LobbyUser.Name == e.UserName && !x.IsSpectator)) return false;
```
Comment (Licho1) [inline Springie/Springie/autohost/AutoHost.cs]: There are several "levels" of spec.
1) lobby spec which can be changed at any time - you used this
2) ingame spec, which cannot be faked. 

I would suggest using ingame spec if game is running.

**Comment by sprunk** [comment]
Using ingame spec where applicable is a good idea, i will look into it.

Currently everything that is needed can be done through buttons in the lobby so !help isn't much needed (but i'll change to allow it anyway).

Instead of allowing votes to fix extreme circumstances it's better to prevent extreme circumstances from existing. For example if laggers don't get resigned then instead of votekicking them, better to change resign to affect laggers ( see https://github.com/ZeroK-RTS/Zero-K/pull/641 ). Non-specs can also still use the commands to deal with any problems.

**Comment by sprunk** [comment]
Now uses ingame spec if spring is running.
Allowed !help, !ring, and !votemove.

**Comment by Licho1** [comment]
pull master manually to hjere to fix auto merge

**Comment by sprunk** [comment]
#557 for fixed auto merge

**Pull request closed without merging**